src/effects/backends/builtin/whitenoiseeffect.cpp

```diff
@@ -50,24 +58,40 @@ void WhiteNoiseEffect::processChannel(
 
     WhiteNoiseGroupState& gs = *pState;
 
-    CSAMPLE drywet = static_cast<CSAMPLE>(m_pDryWetParameter->value());
+    const CSAMPLE drywet = static_cast<CSAMPLE>(m_pDryWetParameter->value());
     RampingValue<CSAMPLE_GAIN> drywet_ramping_value(
             drywet, gs.previous_drywet, engineParameters.framesPerBuffer());
+    auto fade_in_out_ramping = RampingValue<CSAMPLE_GAIN>(
+            enableState == EffectEnableState::Enabling ? 0 : 1,
+            enableState == EffectEnableState::Disabling ? 0 : 1,
+            engineParameters.framesPerBuffer());
 
-    std::uniform_real_distribution<> r_distributor(0.0, 1.0);
+    // Is this relying on implementation details?
+    DEBUG_ASSERT(engineParameters.samplesPerBuffer() ==
+            engineParameters.framesPerBuffer() *
+                    engineParameters.channelCount());
 
-    for (unsigned int i = 0; i < engineParameters.samplesPerBuffer(); i++) {
-        CSAMPLE_GAIN drywet_ramped = drywet_ramping_value.getNext();
+    for (unsigned int frameIndex = 0;
+            frameIndex < engineParameters.framesPerBuffer();
+            frameIndex++) {
+        const CSAMPLE_GAIN drywet_ramped = drywet_ramping_value.getNext();
+        const CSAMPLE_GAIN fade_in_out_gain = fade_in_out_ramping.getNext();
 
-        float noise = static_cast<float>(
-                r_distributor(gs.gen));
+        for (unsigned int sampleInFrameIndex = 0;
+                sampleInFrameIndex < engineParameters.channelCount();
+                sampleInFrameIndex++) {
+            constexpr float normalization_divisor =
+                    static_cast<float>(std::numeric_limits<uint32_t>::max());
+            gs.random_state = nextState(gs.random_state);
+            const float noise = static_cast<float>(gs.random_state) / normalization_divisor;
 
-        pOutput[i] = pInput[i] * (1 - drywet_ramped) + noise * drywet_ramped;
+            const int sampleIndex =
```

Comment:
We use SINT for index in context of sample buffer. That may allow vectorization.

Reply:
Sure, but the primary reason for this loop not being vectorized is because the loop iterations are interdependent. This is due to the use of `RampingValue::getNext()`. Similar thing for the actual xorshift since the states are interdepent there as well. Both implementations could be changed to be vectorizer friendly, but I didn't want to spend that effort yet.

Reply:
We use SINT independent from vectorization.

Reply:
Makes sense, though why the `ptrdiff_t`, We're always indexing into a collection, why use a signed type?

Reply:
In c++ array index can be negative, and GCC has refused to vectorise with unsigned in some cases. 

https://stackoverflow.com/questions/3174850/what-is-the-correct-type-for-array-indexes-in-c